solutions/src/test/java/ru/sberbank/school/task10/ConcurrentScalableThreadPoolTest.java

```diff
@@ -0,0 +1,169 @@
+    private ThreadPool threadPool;
+
+    @Before
+    public void init() {
+        threadPool = new ConcurrentScalableThreadPool(core, max);
+    }
+
+    private Runnable newRunnable(int time, AtomicInteger counter, CountDownLatch latch) {
+        return () -> {
+            try {
+                TimeUnit.MILLISECONDS.sleep(time);
+            } catch (InterruptedException ignore) {
+//                        e.printStackTrace();
+            }
+            synchronized (counter) {
+                System.out.println("Task: " + counter.incrementAndGet()
+                        + " - Thread: " + Thread.currentThread().getName());
+            }
+            latch.countDown();
+        };
+    }
+
+    private Callable<String> newCallable(int time, AtomicInteger counter, CountDownLatch latch) {
+        return () -> {
+            TimeUnit.MILLISECONDS.sleep(time);
+            String res;
+            synchronized (counter) {
+                res = "Task: " + counter.incrementAndGet() + " - Thread: " + Thread.currentThread().getName();
+            }
+            latch.countDown();
+            return res + " " + LocalDateTime.now().toString();
+        };
+    }
+
+    @Test
+    public void normalWork() throws InterruptedException {
+        final AtomicInteger counter = new AtomicInteger(0);
+        Thread.sleep(10);
+        int initialThreadSize = Thread.getAllStackTraces().keySet().size();
+        CountDownLatch latch;
+        int nowThreadSize;
+
+        threadPool.start();
+        Thread.sleep(10);
+        nowThreadSize = Thread.getAllStackTraces().keySet().size();
+        assertEquals(core, nowThreadSize - initialThreadSize);
+
+        latch = new CountDownLatch(25);
+        for (int i = 0; i < 25; i++) {
+            threadPool.execute(newRunnable(50, counter, latch));
+        }
+
+        Thread.sleep(10);
+        nowThreadSize = Thread.getAllStackTraces().keySet().size();
+        assertEquals(max, nowThreadSize - initialThreadSize);
+        latch.await();
+
+        latch = new CountDownLatch(25);
+        for (int i = 0; i < 25; i++) {
+            TimeUnit.MILLISECONDS.sleep(6);
+            threadPool.execute(newRunnable(5, counter, latch));
+        }
+
+        Thread.sleep(10);
+        nowThreadSize = Thread.getAllStackTraces().keySet().size();
+        assertEquals(core, nowThreadSize - initialThreadSize);
+        latch.await();
+
+        assertEquals(50, counter.get());
+
+        threadPool.stopNow();
+        Thread.sleep(10);
+        nowThreadSize = Thread.getAllStackTraces().keySet().size();
+        assertEquals(nowThreadSize, initialThreadSize);
+    }
+
+    @Test
+    public void callableService() throws InterruptedException, ExecutionException {
+        final AtomicInteger counter = new AtomicInteger(0);
+        ArrayList<Future<String>> futureList = new ArrayList<>();
+        Thread.sleep(10);
+        int initialThreadSize = Thread.getAllStackTraces().keySet().size();
```

Comment:
Distance between variable &#39;initialThreadSize&#39; declaration and its first usage is 4, but allowed 3.  Consider making that variable final if you still need to store its value in advance (before method calls that might have side effects on the original value).

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1114914814>*